bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/AuditorPlacementPolicyCheckTaskTest.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.bookkeeper.test.BookKeeperClusterTestCase;
+import org.apache.bookkeeper.test.TestStatsProvider;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Unit test {@link AuditorPlacementPolicyCheckTask}.
+ */
+public class AuditorPlacementPolicyCheckTaskTest extends BookKeeperClusterTestCase {
+    private static final Logger LOG = LoggerFactory
+            .getLogger(AuditorPlacementPolicyCheckTaskTest.class);
+
+    private BookKeeperAdmin admin;
+    private LedgerManager ledgerManager;
+    private LedgerUnderreplicationManager ledgerUnderreplicationManager;
+
+    public AuditorPlacementPolicyCheckTaskTest() {
+        super(3);
+        baseConf.setPageLimit(1);
+        baseConf.setAutoRecoveryDaemonEnabled(false);
+    }
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        final BookKeeper bookKeeper = new BookKeeper(baseClientConf);
+        admin = new BookKeeperAdmin(bookKeeper, NullStatsLogger.INSTANCE, new ClientConfiguration(baseClientConf));
+        LedgerManagerFactory ledgerManagerFactory = bookKeeper.getLedgerManagerFactory();
+        ledgerManager = ledgerManagerFactory.newLedgerManager();
+        ledgerUnderreplicationManager = ledgerManagerFactory.newLedgerUnderreplicationManager();
+    }
+
+    @Test
+    public void testPlacementPolicyCheck() throws BKException, InterruptedException {
+
+        // 1. create ledgers
+        final int numLedgers = 10;
+        List<Long> ids = new LinkedList<Long>();
+        for (int i = 0; i < numLedgers; i++) {
+            LedgerHandle lh = bkc.createLedger(3, 3, BookKeeper.DigestType.CRC32, "passwd".getBytes());
+            ids.add(lh.getId());
+            for (int j = 0; j < 2; j++) {
+                lh.addEntry("testdata".getBytes());
+            }
+            lh.close();
+        }
+
+        // 2. init auditorPlacementPolicyCheckTask
+        final TestStatsProvider statsProvider = new TestStatsProvider();
+        final TestStatsProvider.TestStatsLogger statsLogger = statsProvider.getStatsLogger(AUDITOR_SCOPE);
+        final AuditorStats auditorStats = new AuditorStats(statsLogger);
+
+        AuditorPlacementPolicyCheckTask auditorPlacementPolicyCheckTask = new AuditorPlacementPolicyCheckTask(
+                baseConf, auditorStats, admin, ledgerManager,
+                ledgerUnderreplicationManager, null);
+
+        // 3. placementPolicyCheck
+        auditorPlacementPolicyCheckTask.runTask();
+
+        // 4. verify
+        assertEquals("PLACEMENT_POLICY_CHECK_TIME", 1, ((TestStatsProvider.TestOpStatsLogger)
+                statsLogger.getOpStatsLogger(ReplicationStats.PLACEMENT_POLICY_CHECK_TIME)).getSuccessCount());
+        assertEquals("numOfClosedLedgersAuditedInPlacementPolicyCheck",
+                numLedgers,
+                auditorPlacementPolicyCheckTask.getNumOfClosedLedgersAuditedInPlacementPolicyCheck().get());
```

Comment:
We can also check `numOfLedgersFoundNotAdheringInPlacementPolicyCheck`.

Reply:
addressed done.